src/main/java/org/benetech/servicenet/filters/RequestFilter.java

```diff
@@ -0,0 +1,120 @@
+        throws ServletException, IOException {
+        ContentCachingRequestWrapper wrappedRequest = new ContentCachingRequestWrapper(request);
+        ContentCachingResponseWrapper wrappedResponse = new ContentCachingResponseWrapper(response);
+
+        filterChain.doFilter(wrappedRequest, wrappedResponse);
+
+        Map<String, Object> trace = getTrace(wrappedRequest);
+        Map<String, Object> responseTrace = getResponseTrace(wrappedResponse);
+
+        getBody(wrappedRequest, trace);
+        getResponseBody(wrappedResponse, responseTrace);
+
+        RequestLogger requestLogger = new RequestLogger();
+        requestLogger.setRemoteAddr(Objects.toString(trace.get("remoteAddr"), ""));
+        requestLogger.setRequestUri(Objects.toString(trace.get("requestUri"), ""));
+        requestLogger.setRequestMethod(Objects.toString(trace.get("requestMethod"), ""));
+        requestLogger.setRequestParameters(Objects.toString(trace.get("requestParams"), ""));
+        requestLogger.setRequestBody(Objects.toString(trace.get("body"), ""));
+        requestLogger.setResponseStatus(Objects.toString(responseTrace.get("status"), ""));
+        requestLogger.setResponseBody(Objects.toString(responseTrace.get("body"), ""));
+
+        requestLoggerService.save(requestLoggerMapper.toDto(requestLogger));
+
+        wrappedResponse.copyBodyToResponse();
+    }
+
+    private void getBody(ContentCachingRequestWrapper request, Map<String, Object> trace) {
+        ContentCachingRequestWrapper wrapper = WebUtils.getNativeRequest(request, ContentCachingRequestWrapper.class);
+        if (wrapper != null) {
+            byte[] buf = wrapper.getContentAsByteArray();
+            if (buf.length > 0) {
+                String payload;
+                try {
+                    payload = new String(buf, 0, buf.length, wrapper.getCharacterEncoding());
+                }
+                catch (UnsupportedEncodingException ex) {
+                    payload = "[unknown]";
+                }
+
+                trace.put("body", payload.replaceAll("\"password\":\".+?\"", "\"password\":\"fakePassword\""));
+            }
+        }
+    }
+
+    private void getResponseBody(ContentCachingResponseWrapper response, Map<String, Object> trace) {
+        ContentCachingResponseWrapper wrapper = WebUtils.getNativeResponse(response, ContentCachingResponseWrapper.class);
+        if (wrapper != null) {
+            byte[] buf = wrapper.getContentAsByteArray();
+            if (buf.length > 0) {
+                String payload;
+                try {
+                    payload = new String(buf, 0, buf.length, wrapper.getCharacterEncoding());
+                }
+                catch (UnsupportedEncodingException ex) {
+                    payload = "[unknown]";
+                }
+
+                trace.put("body", payload.replaceAll("\"id_token\" : \".+?\"", "\"id_token\" : \"fakeIdToken\""));
```

Comment:
do we need to filter out tokens?

Reply:
I think it is not as crucial as passwords. I can get rid of it.